packages/SqueakMaps-APIs.package/SMANominatimGeocoder.class/instance/reverse.withZoom..st

```diff
@@ -0,0 +1,12 @@
+as yet unclassified
+reverse: aLatLong withZoom: aZoom
+
+	| json queryUrl | 
+	queryUrl := self reverseUrlFor: aLatLong withZoom: aZoom.
+	json := self contentSource getJsonFrom: queryUrl.
+
+	^ SMAPin new
+			label: (json at: 'display_name' ifAbsent: ['Nix Display name']);
+			latLong: ((json at: 'lon') asNumber) @ ((json at: 'lat') asNumber);
```

Comment:
Hier der Stand des aktuellen Devs: 
```
reverse: aLatLong withZoom: aZoom

	| json | 

	json := Json readFrom: (self reverseUrlFor: aLatLong withZoom: aZoom) retrieveContents contentStream.
	^ json at: 'display_name' 
```
Dort wird also von der Reverse Anfrage nur das Label zurückgegeben, aber die vorher auf dem Mausklick basierende Geokoordinate für die Anzeige des Markers verwendet. Das ist jetzt anders, da die reverse Anfrage mit der berechneten Geokoordinate gestellt wird und wir als Ergebnis einen Pin zurückgeben, der auch die LatLong der Anfrage verwendet und nicht unsere vorher berechnete.

Ich finde es so, wie es nach dem Refactor ist besser, da dann Ortsangabe und Pin übereinstimmen. Würde man aber in die Mitte von Sibirien klicken, wäre das wahrscheinlich nicht so cool, wenn dann 100km weiter der nächstgelegene Punkt angezeigt wird. Eine Möglichkeit wäre, einen Radius um den angeklickten Punkt zu bestimmen und wenn die Distanz zum nächsten erreichbaren Punkt relativ gering ist (kleiner als der Radius) dann zeigt man den nächstgelegenen Punkt an. Andernfalls zeigt man den angeklickten Punkt an, aber das Label des nächstgelegenen Ortes. Weiß aber nicht, ob das nicht etwas over the top ist 😅